File: demo-user-interface/src/main/java/com/demo/user/service/UserService.java
package com.demo.user.service;
import com.demo.pojo.TbUser;
import java.util.List;

public interface UserService {
    /**
     * 查询所有用户
     * @return
     */
    public List<TbUser> findUserList();

    /**
     * 根据用户名查询用户
     * @param username
     * @return
     */
    public TbUser findUser(String username);

    /**
     * 通过年龄段查询用户
     * @param start
     * @param end
     * @return
     */
    public List<TbUser> findUsersByAge(Integer start,Integer end);
}

File: demo-user-web/src/main/webapp/js/service/userService.js
app.service("userService", function ($http) {//构建前端服务层

    this.findList = function () {
        return $http.get('user/findList.do');
    }

    this.findUser = function (username) {
        return $http.get('user/findUser.do?username=' + username);
    }

    this.findUsersByAge = function (start, end) {
        return $http.get('user/findUsersByAge.do?start=' + start + '&end=' + end);
    }
})